### PR TITLE
[feature] Implement the fn:element-with-id function

### PR DESCRIPTION
**Pull request by chris-evolvedbinary**
### Description:

Implemented the `fn:element-with-id` function.

### Reference:

* https://github.com/eXist-db/exist/issues/3635
* https://www.w3.org/TR/xpath-functions-31/#func-element-with-id

### Type of tests:

Added tests: `exist-core/src/test/xquery/xquery3/fnElementWithId.xqm`

---

This open source contribution to the eXist-db project was commissioned by the Office of the Historian, U.S. Department of State, https://history.state.gov/.

### CODE REVIEW

File: exist-core/src/test/xquery/xquery3/fnElementWithId.xqm [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ : You should have received a copy of the GNU Lesser General Public
+ : License along with this library; if not, write to the Free Software
+ : Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ :)
+xquery version "3.1";
+
+module namespace fnewi="http://exist-db.org/xquery/test/function_element_with_id";
+
+declare namespace test="http://exist-db.org/xquery/xqsuite";
+declare namespace x="httpx://x/ns/1.0";
```
Comment (joewiz) [inline exist-core/src/test/xquery/xquery3/fnElementWithId.xqm:27]: This namespace declaration doesn't appear to be used.

Comment (chris-evolvedbinary) [reply exist-core/src/test/xquery/xquery3/fnElementWithId.xqm]: Thank you for catching this.  I removed the unused namespace declaration.

**Comment by joewiz** [comment]
@chris-evolvedbinary How does this PR perform against the relevant XQTS tests?

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunElementWithId.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                    arity(PARAM_ID_REFS_STRING)
+            )
+    );
+
+	public FunElementWithId(final XQueryContext context, final FunctionSignature signature) {
+		super(context, signature);
+	}
+
+    @Override
+	public Sequence eval(Sequence contextSequence, final Item contextItem) throws XPathException {
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/functions/fn/FunElementWithId.java:63]: Why didn't you extend BasicFunction?

Comment (dizzzz) [reply exist-core/src/main/java/org/exist/xquery/functions/fn/FunElementWithId.java:63]: You have probably good reasons!

**Review by dizzzz — APPROVED** [review]

**Comment by chris-evolvedbinary** [comment]
@joewiz: I fixed the failing XQTS tests.

@dizzzz: I changed the function to extend `BasicFunction`.  Thank you for catching this.

**Comment by chris-evolvedbinary** [comment]
All of the `fn:element-with-id` XQTS tests (i.e., `fn-element-with-id-1`, `fn-element-with-id-2`, `fn-element-with-id-3`, `fn-element-with-id-4`, `fn-element-with-id-5`) are now passing.

**Comment by joewiz** [comment]
@chris-evolvedbinary Great! Many thanks!

**Comment by dizzzz** [comment]
@chris-evolvedbinary as a bonus you can leave out the profiling stuff, that is handled by BasicFunction!

**Comment by chris-evolvedbinary** [comment]
@dizzzz: Thank you for pointing out `BasicFunction` profiling.  I removed the redundant profiling code.

**Pull request merged**